### ISSUE TITLE
Make EHBS optional

### ISSUE THREAD

**Issue by Mythos** [Mythos/RimWorld-Diseases-Overhauled#8]
Somehow we should make EHBS optional.
At the moment it is a dependency as some diseases are using the bones/organs of EHBS.

**Comment by Mythos** [comment]
Yeah I saw that, but that means the diseases will never affect bones/organs
added by EHBS.
It would be better to somehow patch those body parts in.

Mystikvm <notifications@github.com> schrieb am So., 11. Nov. 2018, 22:27:

**Comment by Mystikvm** [comment]
Ah, I found most of the additions made by EHBS rather redundant. Also they seemed not very functional. I had pawns with a destroyed upper arm, yet they retained all functionality. Nothing seemed to be tied to those body parts. Same goes for the left and right chest. They are purely cosmetic. Instead of trying to get some functionality into those body parts I just removed them entirely.

I'm for patching them in if you'd like the added complexity, but with my limited knowledge I'd be of little help. I'd just leave them out if I were you and stick with the vanilla bones and organs.

**Comment by Mythos** [comment]
Yeah the useless upper arm.... I saw that after I found a destroyed one. I
mean if the upper arm is gone, how can you have the others still?
This needs to be fixed somehow.

For example for sepsis I'd prefer the complexity of EHBS. Need to consult
the gurus on discord on how to patch that. :-)

Mystikvm <notifications@github.com> schrieb am So., 11. Nov. 2018, 23:01:

**Comment by Mystikvm** [comment]
Just like Mythos, I'm not a modder, so maybe I'm saying dumb things:

> upper arm is actually the description of a module slot.

Then it looks incomplete. In EHBS I found the Bodypart descriptions and it does describe the arm with two module slots. I assume this is for bionics, right? In any case, nowhere I can find any functionality attributes for upper arms or legs. Seems to me that the XML should overwrite the bodypart descriptions in Core with the extra body parts, including any functional tags. Manipulation for upper arms, probably. There are also things like an ulna that don't do anything right now so there are more useless body parts in EHBS right now.

**Comment by EPSIL0N** [comment]
Ulna, Fibula and Patella have the correct tags for manipulation and movement group respectively as far as I can tell. The module slots are a mess.

EHBS is mainly just to allow the completely different mods of DO and CM to run together, seeing as the moduleslots are completely useless to DO, not all of them are even used in CM and may be bad if there is any other mod that allows sicknesses to target any body part similar to what we see in Mythos/RimWorld-Extended-Human-Body-Simulation#5 I'm all for moving the bone definition from EHBS to here and the artificial bone crafting and installation could be moved from CM to here too.

**Comment by Mythos** [comment]
Which bone definitions do you mean?

I would be in favor of moving those module slots to Crafted Mutants where they belong IMHO
Patching them into the body like we should do here to make EHBS optional and patch it in if it's loaded. 

That would make EHBS much cleaner if it only contains new bones/organs

**Comment by Mythos** [comment]
EHBS should still be standalone as apparently people are using it without DO or CM, as it only adds new bones/organs to be destroyed

**Comment by EPSIL0N** [comment]
I'll look again, I believe currently the recipes for some or all of the artificial bones are part of CM and use the EPOE table, possibly the organs too, I'll have to learn if there is a good method to patch body plans instead of simply overwriting them.

**Comment by Mythos** [comment]
Everything is theoretically finished now, but it needs review and some testing.
